src/SFA.DAS.ApprenticeFeedback.Application/Commands/CreateApprenticeFeedback/CreateApprenticeFeedbackHandler.cs

```diff
@@ -33,15 +34,45 @@ public async Task<CreateApprenticeFeedbackResponse> Handle(CreateApprenticeFeedb
             }
 
             var validAttributes = await _apprenticeFeedbackRepository.GetAttributes();
+            var validAttributesIds = validAttributes.Select(a => a.AttributeId).ToList();
+            var providedAttributesIds = request.FeedbackAttributes.Select(a => a.Id).ToList();
+            var invalidAttributesIds = providedAttributesIds.Except(validAttributesIds).ToList();
+            var attributesProvidedNames = GetAttributeNames(request);
+            string invalidAttributeNamesOutput = (invalidAttributesIds.Count != 0) ? CreateInvalidOutput(request, invalidAttributesIds) : string.Empty;
 
             var allValidAttributes = request.FeedbackAttributes.Select(s => s.Id).All(t => validAttributes.Select(t => t.AttributeId).Contains(t));
```

Comment:
I think this line can be removed, as the check has been done above with the invalidAttributeId's list, if that's not empty, then we can log the error and throw the exception.

Following on from that, we can move the `invalidAttributeNamesOutput` inside the if statement and lose the inline if check as it is taken care of by the moving inside the existing if.

It then means we only call that method if we know there is work to be done, where as currently it's called all the time.
The same can be said for the GetAttributeNames